python/lsst/ts/m2gui/display/item_actuator.py

```diff
@@ -0,0 +1,147 @@
+# This file is part of ts_m2gui.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["ItemActuator"]
+
+import numpy as np
+
+from PySide2.QtCore import Qt
+from PySide2.QtWidgets import QGraphicsEllipseItem, QGraphicsTextItem
+from PySide2.QtGui import QPen
+
+from . import Gauge
+
+
+class ItemActuator(QGraphicsEllipseItem):
+    """Actuator item used in the ViewMirror class to show the actuator
+    information.
+
+    Parameters
+    ----------
+    x : `float`
+        Actuator x position on the mirror's view.
+    y : `float`
+        Actuator y position on the mirror's view.
+    diameter : `float` or `int`
+        Diameter of the actuator on the mirror's view.
+    actuator_id : `int`
+        Actuator ID.
+    alias : `str`
+        Alias of the actuator.
+    point_size : `int`
+        Point size of the text.
+
+    Attributes
+    ----------
+    label_id : `PySide2.QtWidgets.QGraphicsTextItem`
+        Label of the actuator ID.
+    """
+
+    # Maximum of the magnitude to decide the color of actuator on the mirror's
+    # view
+    MAX_MAGNITUDE = 700
+
+    def __init__(self, x, y, diameter, actuator_id, alias, point_size):
+        super().__init__(x, y, diameter, diameter)
+
+        self._acutator_id = actuator_id
+        self._alias = alias
+
+        # Magnitude to to decide the color of actuator on the mirror's view
+        self._magnitude = 0
+
+        self.label_id = QGraphicsTextItem(str(actuator_id))
+
+        self._set_default(point_size)
+
+    def _set_default(self, point_size, width_pen=1):
+        """Do the default settings.
+
+        Parameters
+        ----------
+        point_size : `int`
+            Point size of the text.
+        width_pen : `int`, optional
+            Width of the pen. (the default is 1)
+        """
+
+        font = self.label_id.font()
+        font.setPointSize(point_size)
+        self.label_id.setFont(font)
+
+        pen = QPen(Qt.black)
+        pen.setWidth(width_pen)
+        self.setPen(pen)
+
+        self.setBrush(Qt.red)
+
+    def set_position_label_id(self, x, y):
```

Comment:
It would be much better if that can be called from this class, instead of doing calculations outside. Also, you can get bounding rect of text item and center it (e.g. offset it from center position).

Reply:
I would prefer to let the higher-level class (`ViewMirror`) to decide its position. Thanks!